### PR TITLE
384 coverage for burn tensor api int

### PR DESCRIPTION
**Pull request by gonzalesMK**
## Pull Request Template

I implemented some tests for functions with missing coverage. I'm aware there are missing tests still, but I decided to create a first PR to gather early feedback since this is my first contribution. I intend to implement all tests for tensor/

### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Related Issues/PRs

#384  

### Changes / Testing

Included some tests for functions

- [x] arange_device
- [x] arange_step_device
- [x] arange_step_device
- [x] zeros_like
- [x] ones_like
- [x] randoms_like
- [x] one_hot
- [x] var_mean
- [x] var_bias 
- [x] var_mean_bias

### CODE REVIEW

**Review by louisfd — CHANGES_REQUESTED** [review]
Thank you for these new tests! It's always useful to augment our test coverage 😄 
Concerning the `should_panic` tests, I think it's good to check for these edge cases, but I don't think we should expect a specific panic message as this string comparison is too sensible to even minor changes. Just leaving `#[should_panic]` will be fine.

File: burn-tensor/src/tests/ops/gather_scatter.rs [hunk 1]
```diff
@@ -164,4 +164,14 @@ mod tests {
             Data::from([[0.0, 1.0, 0.0], [0.0, 0.0, 4.0]])
         );
     }
+
+    #[test]
+    #[should_panic]
+    fn should_scatter_panic_on_mismatch_of_shapes() {
```
Comment (nathanielsimard) [inline burn-tensor/src/tests/ops/gather_scatter.rs:170]: `should_scatter_panic_on_mismatch_of_shapes` => `scatter_should_panic_on_mismatch_of_shapes`

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
Thanks a lot for this work; it's probably going to help improve the codebase and avoid some bugs! I have some comments before merging, but overall, great job!

**Review by nathanielsimard — APPROVED** [review]
